components/match2/wikis/arenafps/match_legacy.lua

```diff
@@ -0,0 +1,101 @@
+---
+-- @Liquipedia
+-- wiki=halo
+-- page=Module:Match/Legacy
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local p = {}
+
+local Json = require('Module:Json')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+
+local MAX_NUM_PLAYERS = 10
+
+function p.storeMatch(match2, options)
+	if options.storeMatch1 then
+		local match = p._convertParameters(match2)
+
+		match.games = p.storeGames(match, match2)
+
+		return mw.ext.LiquipediaDB.lpdb_match(
+			'legacymatch_' .. match2.match2id,
+			match
+		)
+	end
+end
+
+function p.storeGames(match, match2)
+	local games = ''
+	for gameIndex, game in ipairs(match2.match2games or {}) do
+		game = Table.deepCopy(game)
+		game.opponent1 = match.opponent1
+		game.opponent2 = match.opponent2
+		game.opponent1flag = match.opponent1flag
+		game.opponent2flag = match.opponent2flag
+		game.date = match.date
+		local scores = game.scores or {}
+		if type(scores) == 'string' then
+			scores = Json.parse(scores)
+		end
+		game.opponent1score = scores[1] or 0
+		game.opponent2score = scores[2] or 0
+		local res = mw.ext.LiquipediaDB.lpdb_game(
+			'legacygame_' .. match2.match2id .. gameIndex,
+			game
+		)
+		games = games .. res
+	end
+	return games
+end
+
+function p._convertParameters(match2)
+	local match = Table.deepCopy(match2)
+	for key, _ in pairs(match) do
+		if String.startsWith(key, 'match2') then
+			match[key] = nil
+		end
+	end
+
+	match.staticid = match2.match2id
+
+	-- Handle Opponents
+	local handleOpponent = function (index)
+		local prefix = 'opponent' .. index
+		local opponent = match2.match2opponents[index] or {}
+		local opponentmatch2players = opponent.match2players or {}
+		if opponent.type == 'team' then
+			match[prefix] = mw.ext.TeamTemplate.teampage(opponent.template)
+			match[prefix .. 'score'] = (tonumber(opponent.score) or 0) > 0 and opponent.score or 0
+			local opponentplayers = {}
+			for i = 1, MAX_NUM_PLAYERS do
+				local player = opponentmatch2players[i] or {}
+				opponentplayers['p' .. i] = player.name or ''
+				opponentplayers['p' .. i .. 'flag'] = player.flag or ''
+				opponentplayers['p' .. i .. 'dn'] = player.displayname or ''
+			end
+			match[prefix .. 'players'] = mw.ext.LiquipediaDB.lpdb_create_json(opponentplayers)
```

Comment:
this even needed (rath mentioned it is a 1v1 wiki???)

Reply:
Arena FPS features both team and 1v1 competitions.
(See for example: https://liquipedia.net/arenafps/VAXEE_FOUR_CHALLENGE
https://liquipedia.net/arenafps/IGN_ProLeague/Season_5)